subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/DistributionFactory.java

```diff
@@ -44,10 +45,12 @@
 
 public class DistributionFactory {
     private final Clock clock;
+    private final ScriptFileResolver scriptFileResolver;
     private File distributionBaseDir;
 
-    public DistributionFactory(Clock clock) {
+    public DistributionFactory(Clock clock, ScriptFileResolver scriptFileResolver) {
```

Comment:
❌  `ScriptFileResolver` is only a transitive dependency here. What `DistributionFactory` really needs is a way to produce a `BuildLayoutFactory` should it be needed.

Reply:
Right, only a `BuildLayoutFactory` is need here. It then gets used in `getDefaultDistribution()` which is the most common case. I would not bother adding an indirection to "provide the factory" should it be needed. Instantiation should be cheap. I'll make it simple first, performance tests will tell us if it is worth it.

Reply:
Done